clippy_config/src/conf.rs

```diff
@@ -102,7 +102,9 @@ pub fn sanitize_explanation(raw_docs: &str) -> String {
     // Remove tags and hidden code:
     let mut explanation = String::with_capacity(128);
     let mut in_code = false;
-    for line in raw_docs.lines().map(str::trim) {
+    for line in raw_docs.lines() {
+        let line = line.strip_prefix(' ').unwrap_or(line);
```

Comment:
The indent will still be removed with this change.

Reply:
It removes the single space `///[here]text`, strip_prefix does not remove multiple elements

Reply:
Arf indeed, thought it was stripping as long as it found the character. But then it creates an unbalance in case you have no whitespace character at the beginning of all lines (for whatever reason) and you then have a block.

Reply:
Yeah though this is exactly how the procedural macro did it as well. I think this (there being no space) is a non-issue overall but if this was already fixed in your PR I think that's a better way of doing it, afaik both rustfmt and clippy never catch this likely typo.

Reply:
Which procedural macro?

Also, I think it's about handling the fact that most of the time, you write code after a whitespace  (`/// bla` instead of `///bla`). It's handled in rustdoc by finding out what's the minimum indent and then stripping according to that. Hum, might be worth to do that actually. Gonna update my PR.

Reply:
In the prior PR, `declare_clippy_lint` was turned from a procedural macro into a declarative macro. It had the same logic before as this PR has

Reply:
Ah - didn't realize it'd leave a leading space. The issue was about it being unintentional, and hard to spot, rather than a style choice

Reply:
> Hum, might be worth to do that actually. Gonna update my PR.

I don't think this is worth the effort/code. This is just to generate the Clippy documentation from the lint documentation in the Clippy code base. This code doesn't have to handle all possible doc writing styles people come up with, like `rustdoc` does.

Reply:
Yeah we can't reuse the `fn`s rustdoc uses here unlike for our lints since we only get them as strings, but I think it's fine to assume our docs are reasonably formatted